### DISCUSSION TITLE
Does the AppView not currently enforce image blob size limits or accept types?

### DISCUSSION THREAD

**Discussion by snarfed** [bluesky-social/atproto#2845] [Protocol (atproto)]
Hi all! I recently implemented (well, finished) lexicon schema validation for stored/emitted records and XRPC method input, output, and params, and I've been seeing a trickle of surprising failures on data that the AppView is evidently ok with. One big one: it seems like image blobs are often allowed with types and sizes that don't satisfy `maxSize` or `accept`. Is this known?

A couple specific examples:

1. [This post](https://bsky.app/profile/did:plc:vdcgkdtty6fz7mnwqeqr7cck/post/3kw4r3jklkzj2) has an image embed blob ([getBlob call](https://atproto.brid.gy/xrpc/com.atproto.sync.getBlob?did=did:plc:vdcgkdtty6fz7mnwqeqr7cck&cid=bafkreihzagkzdw6evlkkx5x73nrgamujz6hy5574athumqodcfm4hh3ecq)) with `size` 1858016, which is larger than `maxSize: 1000000`on [`app.bsky.embed.images#image`](https://github.com/bluesky-social/atproto/blob/98d9a158f9c8e7d5f47dd0799fce5ab0c89394a0/lexicons/app/bsky/embed/images.json#L24), but the AppView allowed it and it's visible on the post.
2. [This profile](https://bsky.app/profile/snozomi.misskey.04.si.ap.brid.gy) has `avatar` ([getBlob call](https://atproto.brid.gy/xrpc/com.atproto.sync.getBlob?did=did:plc:6eobpxcs7olbivmow4afmepq&cid=bafkreieb4bevtq4huuqiodukwy35fxnacfevougmujvjtxq4nnqsomvhqi)) with `mime_type: image/webp`, which doesn't satisfy `accept: [image/png, image/jpeg]` on [`app.bsky.actor.profile`](https://github.com/bluesky-social/atproto/blob/27882035e75059f6b167981093d0c2a5aba7cf07/lexicons/app/bsky/actor/profile.json#L26), but the AppView allowed it and it's visible on the profile.

Totally ok if this is expected! I'm just trying to understand how to reconcile it with any lexicon schema validation I do on my end.

Thanks in advance!

(re: https://github.com/snarfed/bridgy-fed/issues/1348, https://github.com/snarfed/bridgy-fed/issues/1349#issuecomment-2384293350)

**Comment by snarfed** [comment]
We're currently adding image processing in https://github.com/snarfed/granary/pull/844 to get aspect ratios, and we'll probably use our own higher limit for images, maybe 5MB, since the appview still doesn't seem to enforce the lexicon's 1M limit.

**Comment by bnewbold** [comment]
Catching up on some old reports!

This one is quite interesting. I think the short answer is that the bsky appview implementation is not being strict enough here; these posts should be rejected.

cc: @devinivy @ericvolp12

**Comment by snarfed** [comment]
The plot thickens. Evidently the Bluesky CDN has an (undocumented?) limit of 10MB or so for images? We get occasional reports of images getting corrupted when they're over that size, eg https://github.com/snarfed/bridgy-fed/issues/2107, https://github.com/snarfed/bridgy-fed/issues/1925.
